File: js backend/student.js
require("dotenv").config();

const mysql=require("mysql2")
const express=require("express")
const cors=require("cors");
const connection=mysql.createConnection({
    host:process.env.host,
    user:process.env.user,
    password:process.env.password,
    database:process.env.database
});

connection.connect((error)=>{
    if(error) throw error;
    console.log("Database connected")
});

const app=express();
app.use(express.json())
app.use(cors())
app.get("/data",(req,res)=>{
    connection.query("select * from credentials",(error,result)=>{
        if(error){
            return res.status(500).json("error in this ")
        }
        res.status(200).json(result)
    })
})

app.get("/get/:user",(req,res)=>{
    const {user}=req.params;
    connection.query("select user,password from credentials where user= ?",[user],(error,result)=>{
        if(error){
            return res.status(500).json("something went wrong")
        }
        res.status(200).json(result)
    });
});

app.post("/data/addt",(req,res)=>{
    const{email,user,password}=req.body;
    if(!email||!user||!password){
        return res.status(400).json("fill the data correctly")
    }
    connection.query("insert into credentials values (?,?,?)",[email,user,password],(error,result)=>{
        if(error){
            return res.status(500).json("faild to insert the data")
        }
        res.status(200).json("sucessfully inserted the data")
    })
})
app.listen(5000,()=>{
    console.log("server is running")
});

File: js backend/student.py
import os
from dotenv import load_dotenv
from flask import Flask, jsonify, request
from flask_cors import CORS
import mysql.connector

# Load environment variables from .env file
load_dotenv()

# Create a Flask application instance
app = Flask(__name__)
# Enable CORS for all routes
CORS(app)

# --- Database Connection ---
try:
    connection = mysql.connector.connect(
        host=os.getenv("host"),
        user=os.getenv("user"),
        password=os.getenv("password"),
        database=os.getenv("database")
    )
    print("Database connected successfully")
except mysql.connector.Error as err:
    print(f"Error: {err}")
    # Exit if the connection fails
    exit(1)

# --- API Endpoints ---

# GET: Fetch all data from the credentials table
@app.route("/data", methods=['GET'])
def get_all_data():
    try:
        cursor = connection.cursor(dictionary=True) # dictionary=True returns rows as dicts
        cursor.execute("SELECT * FROM credentials")
        result = cursor.fetchall()
        cursor.close()
        return jsonify(result), 200
    except mysql.connector.Error as err:
        return jsonify({"error": f"Something went wrong: {err}"}), 500

# GET: Fetch a specific user's data
@app.route("/get/<user>", methods=['GET'])
def get_user_data(user):
    try:
        cursor = connection.cursor(dictionary=True)
        # Use parameterized query to prevent SQL injection
        query = "SELECT user, password FROM credentials WHERE user = %s"
        cursor.execute(query, (user,))
        result = cursor.fetchall()
        cursor.close()
        return jsonify(result), 200
    except mysql.connector.Error as err:
        return jsonify({"error": f"Something went wrong: {err}"}), 500

# POST: Add new credentials to the table
@app.route("/data/addt", methods=['POST'])
def add_data():
    data = request.get_json()
    email = data.get('email')
    user = data.get('user')
    password = data.get('password')

    if not all([email, user, password]):
        return jsonify({"error": "Fill the data correctly"}), 400

    try:
        cursor = connection.cursor()
        query = "INSERT INTO credentials (email, user, password) VALUES (%s, %s, %s)"
        cursor.execute(query, (email, user, password))
        connection.commit()  # Important: Commit changes to the database
        cursor.close()
        return jsonify({"message": "Successfully inserted the data"}), 200
    except mysql.connector.Error as err:
        return jsonify({"error": f"Failed to insert data: {err}"}), 500

# --- Start the Server ---
if __name__ == "__main__":
    app.run(port=5000, debug=True)